File: backend/app/schemas/order.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, field_validator


class OrderCreate(BaseModel):
    device_id: UUID
    buyer_message: str | None = None


class OrderConfirm(BaseModel):
    seller_remark: str | None = None


class OrderCancel(BaseModel):
    cancel_reason: str

    @field_validator("cancel_reason")
    @classmethod
    def validate_reason(cls, v: str) -> str:
        if not v.strip():
            raise ValueError("取消原因不能为空")
        return v.strip()


class DeviceBrief(BaseModel):
    id: UUID
    title: str
    price: float
    image_url: str | None = None

    model_config = {"from_attributes": True}


class UserBrief(BaseModel):
    id: UUID
    username: str
    nickname: str

    model_config = {"from_attributes": True}


class OrderResponse(BaseModel):
    id: UUID
    order_no: str
    device_id: UUID
    buyer_id: UUID
    seller_id: UUID
    price: float
    status: str
    buyer_message: str | None = None
    seller_remark: str | None = None
    confirmed_at: datetime | None = None
    completed_at: datetime | None = None
    cancelled_at: datetime | None = None
    cancel_reason: str | None = None
    created_at: datetime
    updated_at: datetime
    device: DeviceBrief | None = None
    buyer: UserBrief | None = None
    seller: UserBrief | None = None

    model_config = {"from_attributes": True}


class OrderListItem(BaseModel):
    id: UUID
    order_no: str
    device_id: UUID
    buyer_id: UUID
    seller_id: UUID
    price: float
    status: str
    created_at: datetime
    device_title: str | None = None
    device_image_url: str | None = None

    model_config = {"from_attributes": True}

File: frontend/src/api/order.ts
import request from '../utils/request'

export interface OrderCreate {
  device_id: string
  buyer_message?: string
}

export interface OrderDetail {
  id: string
  order_no: string
  device_id: string
  buyer_id: string
  seller_id: string
  price: string
  status: string
  buyer_message: string | null
  seller_remark: string | null
  confirmed_at: string | null
  completed_at: string | null
  cancelled_at: string | null
  cancel_reason: string | null
  created_at: string
  updated_at: string
  device: { id: string; title: string; price: number; image_url: string | null } | null
  buyer: { id: string; username: string; nickname: string } | null
  seller: { id: string; username: string; nickname: string } | null
}

export interface OrderListItem {
  id: string
  order_no: string
  device_id: string
  buyer_id: string
  seller_id: string
  price: string
  status: string
  created_at: string
  device_title: string | null
  device_image_url: string | null
}

export function createOrder(data: OrderCreate) {
  return request.post('/orders', data)
}

export function getMyOrders(params: { role?: string; status?: string; page?: number; size?: number }) {
  return request.get('/orders', { params })
}

export function getOrderDetail(id: string) {
  return request.get(`/orders/${id}`)
}

export function confirmOrder(id: string, seller_remark?: string) {
  return request.patch(`/orders/${id}/confirm`, { seller_remark })
}

export function rejectOrder(id: string) {
  return request.patch(`/orders/${id}/reject`)
}

export function completeOrder(id: string) {
  return request.patch(`/orders/${id}/complete`)
}

export function cancelOrder(id: string, cancel_reason: string) {
  return request.patch(`/orders/${id}/cancel`, { cancel_reason })
}